Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -708,11 +708,13 @@ Optional<TextPosition> VimMotion::get_position(VimEditingEngine& engine)
             // above in get_range normalizes some values which shouldn't be
             // end-exclusive during normal operations.
             bool is_at_start = range.end().column() == 0;
+            auto& line = editor.line(range.end().line());
+
             size_t column = is_at_start ? 0 : range.end().column() - 1;
+            column = min(column, line.length() - (1 - can_go_over_endline));
```

Comment:
```suggestion
            column = min(column, line.length() - !can_go_over_endline);
```

Since `!` maps any integer to `{0, 1}`, that will make it a bit nicer, i think.